[TimescaleDB] Replace TDEngine with TimescaleDB for model monitoring

alxtkr77 opened the pull request:
## Summary
- Add TimescaleDB Helm chart support (StatefulSet, Service, Secret)
- Remove TDEngine chart templates completely
- Enable TimescaleDB as the default TSDB for MLRun CE

## Related Issues
- ML-11627
- CEML-591
- ML-11845

## Changes
- Added `timescaledb` section to `values.yaml` with configurable image, auth, persistence, and resources
- Created TimescaleDB templates: `statefulset.yaml`, `service.yaml`, `secret.yaml`
- Added TimescaleDB helper templates to `_helpers.tpl`
- Removed TDEngine templates directory
- Updated `admin_installation_values.yaml` to reference timescaledb

## Test plan
- [x] Helm lint passed
- [x] Deployed to local Kind cluster with minimal values
- [x] Verified TimescaleDB pod running with PostgreSQL 17 and TimescaleDB extension

## Related PR
https://github.com/mlrun/functions/pull/961

Commit:
[TimescaleDB] Add TimescaleDB support for model monitoring

Add TimescaleDB as an optional TSDB backend for MLRun model monitoring,
providing an alternative to TDEngine.

- Add timescaledb section in values.yaml (disabled by default)
- Add StatefulSet, Service, and Secret templates
- Add helper templates for labels, selectors, and connection string
- Use timescale/timescaledb-ha:pg16-ts2.17.2-all image

Reference: ML-11482

Commit:
[TimescaleDB] Replace TDEngine with TimescaleDB as default TSDB

Remove TDEngine and use TimescaleDB as the sole time-series database
backend for model monitoring.

- Remove TDEngine templates (statefulset, service, configmap)
- Remove TDEngine configuration from values.yaml
- Remove TDEngine helper templates from _helpers.tpl
- Update admin_installation_values.yaml to reference timescaledb
- Enable TimescaleDB by default

Reference: ML-11482

Commit:
[Tests] Add Kind cluster test script for local chart validation

Commit:
[Tests] Fix variable scoping bug in kind-test.sh

Commit:
[TimescaleDB] Address PR review feedback (ML-11627)

- Add urlquery encoding for username/password in connection string helper
  to handle special characters properly
- Pin image tag to specific version pg17.7-ts2.24.0 for reproducible deployments

alxtkr77 commented on charts/mlrun-ce/values.yaml:
```diff
@@ -458,34 +458,36 @@ prometheus-node-exporter:
+    port: 5432
+  auth:
+    username: postgres
+    password: postgres
```
Keeping default password `postgres` for consistency with other CE components (e.g., minio uses `minio123`). This is a Community Edition for development/testing.

alxtkr77 commented on charts/mlrun-ce/templates/timescaledb/secret.yaml:
```diff
@@ -0,0 +1,12 @@
+  name: {{ include "mlrun-ce.timescaledb.fullname" . }}-secret
+  labels:
+    {{- include "mlrun-ce.timescaledb.labels" . | nindent 4 }}
+type: Opaque
```
Skipping - existing CE secrets (mysql-secret, mlpipeline-minio-artifact, s3-auth-secret) don't have security notes. Keeping consistent with project patterns.

alxtkr77 commented on charts/mlrun-ce/templates/_helpers.tpl:
```diff
@@ -277,27 +279,35 @@ If release name contains chart name it will be used as a full name.
+TimescaleDB connection string for MLRun model monitoring
+*/}}
+{{- define "mlrun-ce.timescaledb.connectionString" -}}
+postgresql://{{ .Values.timescaledb.auth.username }}:{{ .Values.timescaledb.auth.password }}@{{ include "mlrun-ce.timescaledb.fullname" . }}:{{ .Values.timescaledb.service.port }}/{{ .Values.timescaledb.auth.database }}
```
Fixed - added `urlquery` encoding for username and password to properly handle special characters like @, :, %, space, etc.

alxtkr77 commented on charts/mlrun-ce/templates/timescaledb/statefulset.yaml:
```diff
@@ -0,0 +1,97 @@
+            failureThreshold: 6
+          resources:
+            {{- toYaml .Values.timescaledb.resources | nindent 12 }}
+  {{- if .Values.timescaledb.persistence.enabled }}
+  volumeClaimTemplates:
+    - metadata:
+        name: data
+        {{- with .Values.timescaledb.persistence.annotations }}
+        annotations:
+          {{- toYaml . | nindent 10 }}
+        {{- end }}
+      spec:
+        accessModes:
+          - {{ .Values.timescaledb.persistence.accessMode | quote }}
+        {{- if .Values.timescaledb.persistence.storageClass }}
+        storageClassName: {{ .Values.timescaledb.persistence.storageClass | quote }}
+        {{- end }}
+        resources:
+          requests:
+            storage: {{ .Values.timescaledb.persistence.size | quote }}
+  {{- end }}
```
Skipping - CE assumes `persistence.enabled: true` (default). Other components like Jupyter deployment have the same pattern. No emptyDir fallback is used anywhere in CE.

alxtkr77 commented on charts/mlrun-ce/values.yaml:
```diff
@@ -458,34 +458,36 @@ prometheus-node-exporter:
-    pullPolicy: Always
-    tag: "3.0.7.1"
+    repository: timescale/timescaledb-ha
+    tag: "pg17"
```
Fixed - updated to `pg17.7-ts2.24.0` for reproducible deployments. Verified by pulling and testing the image.

alxtkr77 commented on charts/mlrun-ce/templates/timescaledb/statefulset.yaml:
```diff
@@ -0,0 +1,97 @@
+            periodSeconds: 10
+            timeoutSeconds: 5
+            failureThreshold: 6
+          resources:
+            {{- toYaml .Values.timescaledb.resources | nindent 12 }}
```
Skipping - defaults are already defined in values.yaml. Other CE components use either hardcoded values in templates or empty `resources: {}`. No component uses inline if/else defaults.

alxtkr77 commented on charts/mlrun-ce/values.yaml:
```diff
@@ -458,34 +458,36 @@ prometheus-node-exporter:
+    database: postgres
+  persistence:
+    enabled: true
+    size: "10Gi"
```
Skipping - no other CE component has comments explaining storageClass behavior. Keeping consistent with project patterns.

alxtkr77 commented on charts/mlrun-ce/values.yaml:
```diff
@@ -458,34 +458,36 @@ prometheus-node-exporter:
+    size: "10Gi"
+    storageClass: ""
+    accessMode: "ReadWriteOnce"
+    annotations:
```
This is consistent with other CE components. All persistence configs in CE use `helm.sh/resource-policy: "keep"` to prevent data loss on helm uninstall - including mlrun-api-pvc (line 123), mlrun-db-pvc (line 177), jupyterNotebook (line 244), and pipelines (line 328). This is intentional behavior for data persistence.

alxtkr77 commented on tests/kind-test.sh:
```diff
@@ -0,0 +1,292 @@
+        --wait \
+        --debug
+
+    rm -f "${values_file}"
```
Good point. However, this is a developer testing script not intended for production CI/CD. The temp file is in /tmp and will be cleaned by the OS. The script uses `set -o errexit` which means it exits on any error anyway. For simplicity, keeping as-is since it's a manual developer tool, not a critical automation script.

assaf758 replied:
Actually, I think we might want to have CI/CD for mlrun-ce at some point, and then this test can run there - so please see if you can address this.

alxtkr77 replied:
Addressed in fff48ed. Added `trap "rm -f '${values_file}'" RETURN` to ensure the temp file is cleaned up when the function exits, whether successfully or due to failure.

alxtkr77 commented on charts/mlrun-ce/templates/_helpers.tpl:
```diff
@@ -277,27 +279,35 @@ If release name contains chart name it will be used as a full name.
+{{/*
+TimescaleDB connection string for MLRun model monitoring
+*/}}
+{{- define "mlrun-ce.timescaledb.connectionString" -}}
+postgresql://{{ .Values.timescaledb.auth.username }}:{{ .Values.timescaledb.auth.password }}@{{ include "mlrun-ce.timescaledb.fullname" . }}:{{ .Values.timescaledb.service.port }}/{{ .Values.timescaledb.auth.database }}
 {{- end }}
```
The connection string helper is intentionally defined but not directly used in chart templates. It follows the same pattern as other CE helpers like `mlrun-ce.minio.service.url` - these are building blocks that get used during actual integration with MLRun components. The helper provides the correct PostgreSQL connection string format that MLRun model monitoring will use when configured. Full integration with MLRun API config is tracked as follow-up work.

alxtkr77 commented on charts/mlrun-ce/values.yaml:
```diff
@@ -458,34 +458,36 @@ prometheus-node-exporter:
-  # Set timezone here, not in taoscfg
-  timezone: "UTC"
+    port: 5432
+  auth:
+    username: postgres
+    password: postgres
+    database: postgres
+  persistence:
+    enabled: true
+    size: "10Gi"
+    storageClass: ""
+    accessMode: "ReadWriteOnce"
+    annotations:
+      helm.sh/resource-policy: "keep"
   resources:
-  storage:
-    className: ""
-    dataSize: "10Gi"
-    logSize: "1Gi"
-  nodeSelectors:
-    taosd:
-  clusterDomainSuffix: ""
-  taoscfg:
-    CLUSTER: "0"
-    TAOS_REPLICA: "1"
+    requests:
+      memory: "256Mi"
+      cpu: "100m"
+    limits:
+      memory: "1Gi"
+      cpu: "1"
+  nodeSelector: {}
+  tolerations: []
```
Skipping - this is consistent with CE project patterns. Other component sections in values.yaml (nuclio, mlrun, minio, kafka, spark-operator, etc.) don't have inline documentation comments either. CE relies on defaults being self-explanatory and users referencing component documentation for advanced configuration. Adding docs only to timescaledb would create inconsistency.

Commit:
[Tests] Ensure temp file cleanup in kind-test.sh (ML-11627)

Use trap with RETURN signal to ensure temp values file is cleaned up
even if helm command fails, making the script suitable for CI/CD use.
